🤔 OK, something seems to be up with the Windows build agent. @TravisEz13 have you seen this happen before? 
```
##[section]Starting: Download build artifacts
==============================================================================
Task         : Download build artifacts
Description  : Download files that were saved as artifacts of a completed build
Version      : 0.153.1
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/download-build-artifacts
==============================================================================
Downloading artifacts for build: 25058
Linked artifacts count:  3
Downloading artifact build from: https://powershell.visualstudio.com//2972bb5c-f20c-4a60-8bd9-00ffe9987edc/_apis/build/builds/25058/artifacts?artifactName=build&$format=zip
Downloading artifact testResults from: https://powershell.visualstudio.com//2972bb5c-f20c-4a60-8bd9-00ffe9987edc/_apis/build/builds/25058/artifacts?artifactName=testResults&$format=zip
Downloading artifact xunit from: https://powershell.visualstudio.com//2972bb5c-f20c-4a60-8bd9-00ffe9987edc/_apis/build/builds/25058/artifacts?artifactName=xunit&$format=zip
Skipped processing item 
Skipped processing item 
Skipped processing item 
Successfully downloaded artifacts to D:\a\1\a
Total Files: 1, Processed: 0, Skipped: 1, Failed: 0, Download time: 0.015 secs, Download size: 0Bytes

D:\a\_tasks\DownloadBuildArtifacts_a433f589-fce1-4460-9ee6-44a624aeb1fb\0.153.1\node_modules\artifact-engine\Engine\artifactEngine.js:157
    throw err;
    ^
Failed to extract package with error Error: ENOENT: no such file or directory, open 'D:\a\1\a\build.zip'
Total Files: 1, Processed: 0, Skipped: 1, Failed: 0, Download time: 0.015 secs, Download size: 0Bytes
Total Files: 1, Processed: 0, Skipped: 1, Failed: 0, Download time: 0.015 secs, Download size: 0Bytes
##[error]artifact-engine: unhandled rejection Failed to extract package with error Error: ENOENT: no such file or directory, open 'D:\a\1\a\build.zip'
##[error]Unhandled: undefined
##[error]Unhandled: undefined
##[error]artifact-engine: unhandled exception Failed to extract package with error Error: ENOENT: no such file or directory, open 'D:\a\1\a\build.zip'
##[error]Exit code 7 returned from process: file name 'C:\agents\2.152.1\externals\node\bin\node.exe', arguments '"D:\a\_tasks\DownloadBuildArtifacts_a433f589-fce1-4460-9ee6-44a624aeb1fb\0.153.1\main.js"'.
##[section]Finishing: Download build artifacts
```

--- a/src/System.Management.Automation/engine/parser/Parser.cs
+++ b/src/System.Management.Automation/engine/parser/Parser.cs
@@ -4514,8 +4514,11 @@ namespace System.Management.Automation.Language
                 }
             }
 
-            if (token.Kind == TokenKind.Identifier)
+            if (token.Kind == TokenKind.Identifier
+                || token.TokenFlags.HasFlag(TokenFlags.Keyword))
             {
+                token.TokenFlags = TokenFlags.None;
+
                 SkipToken();
                 var functionDefinition = MethodDeclarationRule(token, className, staticToken != null) as FunctionDefinitionAst;
 

--- a/test/powershell/Language/Classes/Scripting.Classes.BasicParsing.Tests.ps1
+++ b/test/powershell/Language/Classes/Scripting.Classes.BasicParsing.Tests.ps1
@@ -416,6 +416,34 @@ Describe 'Property Attributes Test' -Tags "CI" {
         It "second value should be b" { $v.ValidValues[1] | Should -Be 'b' }
 }
 
+Describe 'Method Name Test' -Tags "CI" {
+    It 'Permits class methods to be named after keywords' {
+        $ClassDefinition = @'
+            try {
+                class TestMethodNames : IDisposable {
+                    [string] Begin() { return "Begin" }
+
+                    [string] Process() { return "Process" }
+
+                    [string] End() { return "End" }
+
+                    hidden $Data = "Secrets"
+
+                    Dispose() {
+                        $this.Data = [string]::Empty
+                    }
+                }
+
+                $a = [TestMethodNames]::new()
+            }
+            finally {
+                $a.Dispose()
+            }
+'@
+        { ExecuteCommand $ClassDefinition } | Should -Not -Throw
+    }
+}
+
 Describe 'Method Attributes Test' -Tags "CI" {
         class C { [Obsolete("aaa")][int]f() { return 1 } }
 

--- a/test/powershell/Language/Classes/Scripting.Classes.BasicParsing.Tests.ps1
+++ b/test/powershell/Language/Classes/Scripting.Classes.BasicParsing.Tests.ps1
@@ -417,30 +417,31 @@ Describe 'Property Attributes Test' -Tags "CI" {
 }
 
 Describe 'Method Name Test' -Tags "CI" {
+    BeforeAll {
+        class TestMethodNames : IDisposable {
+            [string] Begin() { return "Begin" }
+
+            [string] Process() { return "Process" }
+
+            [string] End() { return "End" }
+
+            hidden $Data = "Secrets"
+
+            Dispose() {
+                $this.Data = [string]::Empty
+            }
+        }
+    }
     It 'Permits class methods to be named after keywords' {
-        $ClassDefinition = @'
-            try {
-                class TestMethodNames : IDisposable {
-                    [string] Begin() { return "Begin" }
+        $Object = [TestMethodNames]::new()
 
-                    [string] Process() { return "Process" }
+        $Object.Begin() | Should -BeExactly 'Begin'
+        $Object.Process() | Should -BeExactly 'Process'
+        $Object.End() | Should -BeExactly 'End'
 
-                    [string] End() { return "End" }
-
-                    hidden $Data = "Secrets"
-
-                    Dispose() {
-                        $this.Data = [string]::Empty
-                    }
-                }
-
-                $a = [TestMethodNames]::new()
-            }
-            finally {
-                $a.Dispose()
-            }
-'@
-        { ExecuteCommand $ClassDefinition } | Should -Not -Throw
+        $Object.Data | Should -BeExactly 'Secrets'
+        $Object.Dispose()
+        $Object.Data | Should -BeNullOrEmpty
     }
 }
 

